### ISSUE TITLE
[Feature]: With New-AzKeyVault cmdlet it is not possible to set createMode propertie

### ISSUE THREAD

**Issue by vigor-vavan** [Azure/azure-powershell#21482]
### Description of the new feature

New version of built in policy 'Azure Key Vault should have firewall enabled' is now GA.(version 3.1.0)
![image](https://user-images.githubusercontent.com/104625529/230894550-4462782b-e4a3-4bf7-be8c-537a020b7c6d.png)

It is not possible to create new Key Vault and to be compliant with this new policy that is part of many initiatives like : Azure Security Benchmark, CMMC Level 3 ,FedRAMP High...

New-AzKeyVault cmdlet should have possibility to set parameter createMode

**Comment by wyunchi-ms** [comment]
Thanks for your feedback! We will try to add this feature.

**Comment by isra-fel** [comment]
Ref:

https://github.com/Azure/azure-powershell/blob/37ed9f20d422d048246a7fe1c5b349064f727106/src/KeyVault/KeyVault.Sdk/README.md?plain=1#L26

https://github.com/Azure/azure-rest-api-specs/blob/main/specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2022-07-01/keyvault.json#L1272

**Comment by evelyn-ys** [comment]
@jlichwa for awareness

**Comment by jlichwa** [comment]
@vigor-vavan can you clarify, all creation is using create mode, there is separate command for recovery https://learn.microsoft.com/en-us/cli/azure/keyvault?view=azure-cli-latest#az-keyvault-recover

**Comment by vigor-vavan** [comment]
New-AzKeyVault cmdlet doesn`t have parameter createMode.

Listed parameteres are :
New-AzKeyVault
   [-Name] <String>
   [-ResourceGroupName] <String>
   [-Location] <String>
   [-EnabledForDeployment]
   [-EnabledForTemplateDeployment]
   [-EnabledForDiskEncryption]
   [-EnablePurgeProtection]
   [-EnableRbacAuthorization]
   [-SoftDeleteRetentionInDays <Int32>]
   [-PublicNetworkAccess <String>]
   [-Sku <String>]
   [-Tag <Hashtable>]
   [-NetworkRuleSet <PSKeyVaultNetworkRuleSet>]
   [-DefaultProfile <IAzureContextContainer>]
   [-WhatIf]
   [-Confirm]
   [-SubscriptionId <String>]
   [<CommonParameters>]

maybe that parameter is hidden but w as you can see from policy compliance screenshot current value is '--' and all our previously created key vaults are now non compliant with the same reason for non compliance.

**Comment by jlichwa** [comment]
@vigor-vavan New-AzKeyVault is only to create new vault ( createMode default). If you want to use createMode=recover you use completely separate command Undo-AzKeyVaultRemoval as I provided above. So there are 2 commands instead one with REST API for better customer experience as we designed based on customer feedback.

Are you saying that if you create key vault using New-AzKeyVault with firewall enabled it makes you vault non-compliant?
If that is the case then there is an issue in Policy.
